tensorflow_quantum/python/optimizers/rotosolve_minimizer.py

```diff
@@ -0,0 +1,273 @@
+        'num_objective_evaluations',
+        # The total number of objective
+        # evaluations performed.
+        'position',
+        # A tensor containing the last argument value found
+        # during the search. If the search converged, then
+        # this value is the argmin of the objective function.
+        # A tensor containing the value of the objective from
+        # previous iteration
+        'objective_value_previous_iteration',
+        # Save the evaluated value of the objective function
+        # from the previous iteration
+        'objective_value',
+        # A tensor containing the value of the objective
+        # function at the `position`. If the search
+        # converged, then this is the (local) minimum of
+        # the objective function.
+        'tolerance',
+        # Define the stop criteria. Iteration will stop when the
+        # objective value difference between two iterations is
+        # smaller than tolerance
+        'solve_param_i',
+        # The parameter index where rotosolve is currently
+        # modifying. Reserved for internal use.
+    ])
+
+
+def _get_initial_state(initial_position, tolerance, expectation_value_function):
+    """Create RotosolveOptimizerResults with initial state of search."""
+    init_args = {
+        "converged": tf.Variable(False),
+        "num_iterations": tf.Variable(0),
+        "num_objective_evaluations": tf.Variable(0),
+        "position": tf.Variable(initial_position),
+        "objective_value": tf.Variable(0.),
+        "objective_value_previous_iteration": tf.Variable(0.),
+        "tolerance": tolerance,
+        "solve_param_i": tf.Variable(0)
+    }
+    return RotosolveOptimizerResults(**init_args)
+
+
+def minimize(expectation_value_function,
+             initial_position,
+             tolerance=1e-5,
+             max_iterations=50,
+             name=None):
+    """Applies the rotosolve algorithm.
+
+    The rotosolve algorithm can be used to minimize a linear combination
+
+    of quantum measurement expectation values. See the following paper:
+
+    [arXiv:1903.12166](https://arxiv.org/abs/1903.12166), Ken M. Nakanishi.
+    [arXiv:1905.09692](https://arxiv.org/abs/1905.09692), Mateusz Ostaszewski.
+
+    ### Usage:
+
+    The following example demonstrates the Rotosolve optimizer attempting
+     to find the minimum for two qubit ansatz expectation value.
+
+    Here is an example of optimize a function which consists summation of
+    a few sinusoids.
+
+    >>> import tensorflow_quantum as tfq
+    >>> import numpy as np
+    >>> import tensorflow as tf
+    >>> n = 10  # Number of sinusoids
+    >>> coefficient = tf.random.uniform(shape=[n])
+    >>> min_value = -tf.math.reduce_sum(tf.abs(coefficient))
+    >>> func = lambda x:tf.math.reduce_sum(tf.sin(x) * coefficient)
+    >>> # Optimize the function with rotosolve, start with random parameters
+    >>> result = tfq.optimizers.rotosolve_minimizer.minimize( \

```

Comment:
Don't need the `\` can just do `result = tfq.optimizers.rotosolve_minimizer.minimize(`

Reply:
Also shouldn't the function call be `tfq.optimizers.rotosolve_minimize(` and not `tfq.optimizers.rotosolve_minimizer.minimize(` ?

Reply:
Fixed both in e9da877e4b3be1d187164f92a2fcf6aba0def909